app/uk/gov/hmrc/incorporatedentityidentification/connectors/RegisterWithMultipleIdentifiersConnector.scala

```diff
@@ -111,7 +111,7 @@ object RegisterWithMultipleIdentifiersHttpParser {
             case JsSuccess(safeId, _) => RegisterWithMultipleIdentifiersSuccess(safeId)
             case _: JsError => throw new InternalServerException(s"Invalid JSON returned on Register API: ${response.body}")
           }
-        case _ => RegisterWithMultipleIdentifiersFailure(response.status, response.body)
+        case _ => RegisterWithMultipleIdentifiersFailure(response.status, Json.obj("failures" -> response.json).toString)
```

Comment:
Here we need to read the json and check if multiple failures are returned or if only one is returned and then return the failures as an Array back to the controller. Maybe something like this:
`case _ =>
          if (response.json.as[JsObject].keys.contains("failures")) {
            (response.json \ "failures").validate[Array[Failures]] match {
              case JsSuccess(failures, _) => RegisterWithMultipleIdentifiersFailure(response.status, failures)
              case _: JsError => throw new InternalServerException(s"Invalid JSON returned on Register API: ${response.body}")
            }
          }
          else {
            response.json.validate[Failures] match {
              case JsSuccess(failure, _) => RegisterWithMultipleIdentifiersFailure(response.status, Array(failure))
              case _: JsError => throw new InternalServerException(s"Invalid JSON returned on Register API: ${response.body}")
            }
          }`

I update the RegFailure model and created a new one:
`case class RegisterWithMultipleIdentifiersFailure(status: Int, body: Array[Failures]) extends RegisterWithMultipleIdentifiersResult

  case class Failures(code: String, reason: String)

  implicit val format: OFormat[Failures] = Json.format[Failures]`

And then in the controller we need to pass the errors back to the frontend:
` case RegisterWithMultipleIdentifiersFailure(_, body) =>
            Ok(Json.obj(
              "registration" -> Json.obj(
                "registrationStatus" -> "REGISTRATION_FAILED",
              "failures" -> body)))
        }`